pkg/cmd/util/clientcmd/factory.go

```diff
@@ -452,6 +462,16 @@ func getPorts(spec api.PodSpec) []string {
 	return result
 }
 
+func getProtocols(spec api.PodSpec) map[string]string {
+	result := make(map[string]string)
+	for _, container := range spec.Containers {
+		for _, port := range container.Ports {
+			result[strconv.Itoa(int(port.ContainerPort))] = string(port.Protocol)
```

Comment:
> Aren't I allowed to duplicate a port.ContainerPort value with different protocols?

This was added because you weren't able to expose multiple ports at all. One step at a time thing.